emoji-substitution/substitute.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is responsible for performing the logic of replacing
+ * all occurrences of each mapped word with its emoji counterpart.
+ */
+
+// emojiMap.js defines the 'sortedEmojiMap' variable.
+// Referenced here to reduce confusion.
+const emojiMap = sortedEmojiMap;
+
+/*
+ * For efficiency, create a word --> search RegEx Map too.
+ */
+let regexs = new Map();
+for (let word of emojiMap.keys()) {
+  // We want a global, case-insensitive replacement.
+  // @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+  regexs.set(word, new RegExp(word, 'gi'));
```

Comment:
FWIW, I think the current behavior here is fine. I read your description in the PR when you first made it, and agree that it's kind of fun the way it is now. It makes the add-on a kind of rebus-generator :).